doc/user/content/self-managed-deployments/upgrading/_index.md

```diff
@@ -0,0 +1,271 @@
+instance(s) to the **APP Version** of the Operator. To find the version of your
+currently deployed Materialize Operator:
+
+```shell
+helm list -n materialize
+```
+
+You will use the returned **App Version** for the updated `environmentdImageRef`
+value. Specifically, for your Materialize instance(s), set
+`environmentdImageRef` value to use the new version:
+
+```
+spec:
+  environmentdImageRef: docker.io/materialize/environmentd:<app_version>
+```
+
+To minimize unexpected downtime and avoid connection drops at critical
+periods for your application, the upgrade process involves two steps:
+
+- First, stage the changes (update the `environmentdImageRef` with the new
+  version) to the Materialize custom resource. The Operator watches for changes
+  but does not automatically roll out the changes.
+
+- Second, roll out the changes by specifying a new UUID for `requestRollout`.
+
+### Stage the Materialize instance version change
+
+To stage the Materialize instances version upgrade, update the
+`environmentdImageRef` field in the Materialize custom resource spec to the
+compatible version of your currently deployed Materialize Operator.
+
+To stage, but **not** rollout, the Materialize instance version upgrade, you can
+use the `kubectl patch` command; for example, if the **App Version** is {{< self-managed/versions/get-latest-version >}}:
+
+```shell
+kubectl patch materialize <instance-name> \
+  -n <materialize-instance-namespace> \
+  --type='merge' \
+  -p "{\"spec\": {\"environmentdImageRef\": \"docker.io/materialize/environmentd:{{< self-managed/versions/get-latest-version >}}\"}}"
+```
+
+{{< note >}}
+Until you specify a new `requestRollout`, the Operator watches for updates but
+does not roll out the changes.
+{{< /note >}}
+
+
+### Applying the changes via `requestRollout`
+
+To apply chang Materialize instance upgrade, you must update the `requestRollout` field in the Materialize custom resource spec to a new UUID.
+Be sure to consult the [Rollout Configurations](#rollout-configuration) to ensure you've selected the correct rollout behavior.
+```shell
+# Then trigger the rollout with a new UUID
+kubectl patch materialize <instance-name> \
+  -n <materialize-instance-namespace> \
+  --type='merge' \
+  -p "{\"spec\": {\"requestRollout\": \"$(uuidgen)\"}}"
+```
+
+### Staging and applying in a single command
+
+Although separating the staging and rollout of the changes into two steps can
+minimize unexpected downtime and avoid connection drops at critical periods, you
+can, if preferred, combine both operations in a single command
+
+```shell
+kubectl patch materialize <instance-name> \
+  -n materialize-environment \
+  --type='merge' \
+  -p "{\"spec\": {\"environmentdImageRef\": \"docker.io/materialize/environmentd:{{< self-managed/versions/get-latest-version >}}\", \"requestRollout\": \"$(uuidgen)\"}}"
+```
+
+### Using YAML Definition
+
+Alternatively, you can update your Materialize custom resource definition directly:
```

Comment:
Q: Just wondering ... for this (and kind upgrade I suppose) ... should we have it broken into two parts as well? ... One to stage and the other to requestRollout?